### PR TITLE
Added some new questions to JavaScript, topic: arrays

### PR DESCRIPTION
**Pull request by J-ATJ**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

### CODE REVIEW

File: src/data/javascript-quiz.ts [hunk 1]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+  },
+  {
+    Question:
+      "In JavaScript, given the array const arr = ['a', 'b', 'c', 'd', 'e'], what returns ['c', 'd', 'e'] ?",
+    Answer: "arr.slice(2,5)",
+    Distractor1: "arr.fill(c, d, e)",
+    Distractor2: "arr.keys(c, d, e)",
+    Distractor3: "arr.remove(0, 1)",
+    Explanation:
+      "The slice() method returns a shallow copy of a portion of an array into a new array object selected from start to end (end not included)",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1307]:
```suggestion
      "The slice() method has two optional parameters which represent the starting and ending index for the array. This method returns a new shallow copy of the sliced array.",
```

File: src/data/javascript-quiz.ts [hunk 2]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+    Link: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
+  },
+  {
+    Question:
+      "In JavaScript, given the array const n = [22, -4, 99, 103, -30, 172], what is the result of n.find(x => x > 50) ?",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1312]:
```suggestion
      "In JavaScript, what is the result for [22, -4, 99, 103, -30, 172].find(x => x > 50)?",
```

File: src/data/javascript-quiz.ts [hunk 3]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+    Answer: "99",
+    Distractor1: "{99, 103, 172}",
+    Distractor2: "undefined",
+    Distractor3: "[99, 103, 172]",
+    Explanation:
+      "The find() method returns the first element in the provided array that satisfies the provided testing function.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1318]:
```suggestion
      "The find() method will look through the array and return the first element that passes the callback function.",
```

File: src/data/javascript-quiz.ts [hunk 4]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+    Link: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find",
+  },
+  {
+    Question:
+      "In JavaScript, given an array, what method takes an integer value and returns the item at that index ?",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1323]:
```suggestion
      "In JavaScript, what array method takes an integer value and returns the element at that index?",
```

File: src/data/javascript-quiz.ts [hunk 5]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+    Answer: ".at()",
+    Distractor1: ".value()",
+    Distractor2: ".flat()",
+    Distractor3: ".some()",
+    Explanation:
+      "The at() method takes an integer value and returns the item at that index, allowing for positive and negative integers.",
```
Comment (jdwilkin4) [inline src/data/javascript-quiz.ts:1329]:
```suggestion
      "The at() method takes a positive or negative integer and returns the index for that element.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

File: src/data/javascript-quiz.ts [hunk 6]
```diff
@@ -1274,6 +1275,60 @@ const javascriptQuiz = [
...
+    Link: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
+  },
+  {
+    Question: "In JavaScript, which is a correct way to concatenate 3 arrays?",
+    Answer: "const newArr = arr1.concat(arr2, arr3)",
+    Distractor1: "const newArr = join(arr1, arr2, arr3)",
+    Distractor2: "const newArr = combine[arr1, arr2, arr3]",
+    Distractor3: "const newArr = arr1.join[arr2, arr3]",
+    Explanation:
+      "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
```
[the remark this answers is not in the document]
Comment (J-ATJ) [reply src/data/javascript-quiz.ts:1296]: Okok Since English is not my native language I wanted to be very clear and overlooked that!

**Review by J-ATJ — COMMENTED** [review]
I'm going to improve my English for the next PR! Happy to contribute to this project. 😄

**Review by jdwilkin4 — APPROVED** [review]
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**